tests/unit/component/partitioned_vector_exclusive_scan.cpp

```diff
@@ -0,0 +1,360 @@
+    hpx::partitioned_vector<T> in(size, in_policy);
+    iota_vector(in, T(1));
+
+    hpx::partitioned_vector<T> out(size, out_policy);
+
+    std::vector<T> ver(in.size());
+    std::iota(ver.begin(), ver.end(), T(1));
+    T val(0);
+
+    hpx::parallel::v1::detail::sequential_exclusive_scan(
+        ver.begin(), ver.end(), ver.begin(), val, opt<T>());
+
+    //sync
+    exclusive_scan_algo_tests_segmented_out_with_policy<T>(
+        size, in_policy, out_policy, in, out, ver, seq);
+    exclusive_scan_algo_tests_segmented_out_with_policy<T>(
+        size, in_policy, out_policy, in, out, ver, par);
+
+    //async
+    exclusive_scan_algo_tests_segmented_out_with_policy_async<T>(
+        size, in_policy, out_policy, in, out, ver, seq(task));
+    exclusive_scan_algo_tests_segmented_out_with_policy_async<T>(
+        size, in_policy, out_policy, in, out, ver, par(task));
+}
+
+template <typename T, typename DistPolicy>
+void exclusive_scan_tests_inplace_with_policy(std::size_t size,
+    DistPolicy const& policy)
+{
+    using namespace hpx::parallel;
+    using hpx::parallel::task;
+
+    // setup verification vector
+    std::vector<T> ver(size);
+    std::iota(ver.begin(), ver.end(), T(1));
+    T val(0);
+
+    hpx::parallel::v1::detail::sequential_exclusive_scan(
+        ver.begin(), ver.end(), ver.begin(), val, opt<T>());
+
+    //sync
+    exclusive_scan_algo_tests_inplace_with_policy<T>(size, policy, ver, seq);
+    exclusive_scan_algo_tests_inplace_with_policy<T>(size, policy, ver, par);
+
+    //async
+    exclusive_scan_algo_tests_inplace_with_policy_async<T>(
+        size, policy, ver, seq(task));
+    exclusive_scan_algo_tests_inplace_with_policy_async<T>(
+        size, policy, ver, par(task));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+void exclusive_scan_tests()
+{
+    std::size_t const length = 1000000;
+
+    std::vector<hpx::id_type> localities = hpx::find_all_localities();
+
+    exclusive_scan_tests_with_policy<T>(length, hpx::container_layout);
+    exclusive_scan_tests_with_policy<T>(length, hpx::container_layout(3));
+    exclusive_scan_tests_with_policy<T>(length, hpx::container_layout(3, localities));
+    exclusive_scan_tests_with_policy<T>(length, hpx::container_layout(localities));
+
+    exclusive_scan_tests_with_policy<T>(1000, hpx::container_layout(1000));
+
+    exclusive_scan_tests_segmented_out_with_policy<T>(length,
+        hpx::container_layout(localities), hpx::container_layout(localities));
+
+    exclusive_scan_tests_segmented_out_with_policy<T>(length,
+        hpx::container_layout(localities), hpx::container_layout(10));
+
+    exclusive_scan_tests_inplace_with_policy<T>(length,
+        hpx::container_layout(localities));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+int main()
+{
+    std::vector<hpx::id_type> localities = hpx::find_remote_localities();
```

Comment:
Yes those should be distributed to test the version of the algorithm that use the locality of the data. (i.e. scan is performed on partition that owns the data instead of the root partition)